### ISSUE TITLE
Deprecation warnings should not log to console in production

### ISSUE THREAD

**Issue by scitech** [CMSgov/design-system#595]
**Description**

When upgrading the design system in my project I encountered a lot of error-level console messages due to deprecated components.

I propose that the design system follow the convention used by libraries like React where deprecation warnings are warning-level and are only visible in development. So instead of:
```
console.error('<Select> is deprecated');
```

It could be:
```
if (process.env.NODE_ENV !== 'production') {
  console.warn('<Select> is deprecated');
}
```

Because it's wrapped in a `process.env.NODE_ENV` check, application build tools would be able to strip out the `console.warn` call when building for production. This way design system application developers will be able to see and fix the warnings, the production bundle will be slightly smaller, and applications will have less noise in our production console.

**Steps to reproduce the issue**

1. Use a `<Select>` component

**Comment by bernardwang** [comment]
Ah thanks for the issue, this is a big oversight, we can get it in for the next release.

**Comment by bernardwang** [comment]
Tracked by https://jira.cms.gov/browse/WNMGDS-346

**Comment by bernardwang** [comment]
Should be fixed in v3.5.0
